app/vmselect/promql/exec.go

```diff
@@ -63,6 +70,16 @@ func Exec(qt *querytracer.Tracer, ec *EvalConfig, q string, isFirstPointOnly boo
 		return nil, err
 	}
 
+	if *disableImplicitConversion || *logImplicitConversion {
+		complete := isSubQueryComplete(e, false)
+		if !complete && *logImplicitConversion {
```

Comment:
Might wanna check disableImplicitConversion first, since the error message will also be printed in the log. And with both logImplicitConversion and disableImplicitConversion enabled, the log will be duplicated.